hasher-matcher-actioner/src/OpenMediaMatch/static/openapi/openapi.yaml

```
@@ -0,0 +1,1767 @@
+            algorithm = preferredAlgorithm
+          } else {
+            // Pick first available
+            for k := range hashResponse {
+              algorithm = k
+              break
+            }
+          }
+          
+          matchingRequest := map[string]string{
+            "signal":      hashResponse[algorithm],
+            "signal_type": algorithm,
+          }
+          
+          // 3. Find matches
+          return findMatches(matchingRequest)
+        }
+        ```
+        
+        ```swift
+        // Swift: Hash → Transform → Match
+        func hashAndMatch(data: Data, preferredAlgorithm: String? = nil) async throws -> Any {
+          // 1. Hash content (returns dynamic dictionary)
+          let hashResponse: [String: String] = try JSONDecoder().decode([String: String].self, from: hashData)
+          
+          // 2. Transform for matching (manual extraction)
+          let algorithms = Array(hashResponse.keys)
+          let algorithm = preferredAlgorithm.flatMap { algorithms.contains($0) ? $0 : nil } ?? algorithms.first!
+          
+          let matchingRequest = [
+            "signal": hashResponse[algorithm]!,
+            "signal_type": algorithm
+          ]
+          
+          // 3. Find matches  
+          return try await findMatches(matchingRequest)
+        }
+        ```
+
+    MatchingRequest:
+      type: object
+      required: [signal, signal_type]
+      properties:
+        signal:
+          type: string
+          description: |
+            Hash value extracted from HashResponse.
+            **Source**: Must be manually extracted from hashing API response.
+          example: "facd8bcb2a49bcebdec1985298d5fe84bcd006c187c598c720c3c087b3fdb318"
+        signal_type:
+          type: string
+          description: |
+            Algorithm name (key from HashResponse).
+            **Source**: Must match a key from the hashing API response.
+            
+            **Common Values**: pdq, md5, sha1, sha256, sha512, video_md5, phash, dhash, ahash
+            **Note**: Available types depend on content type and HMA configuration.
+          enum: [pdq, md5, sha1, sha256, sha512, video_md5, phash, dhash, ahash, tmk]
+          example: "pdq"
+      example:
+        signal: "facd8bcb2a49bcebdec1985298d5fe84bcd006c187c598c720c3c087b3fdb318"
+        signal_type: "pdq"
+      x-transformation-example: |
+        // Step 1: Get hashing response
+        const hashResponse = await hashContent(file);
+        // Result: { "pdq": "...", "md5": "..." }
+        
+        // Step 2: Manual transformation for matching
+        const algorithm = "pdq"; // or Object.keys(hashResponse)[0]
+        const matchingRequest: MatchingRequest = {
+          signal: hashResponse[algorithm],
+          signal_type: algorithm
+        };
+        
+        // Step 3: Use in matching API
+        const matches = await findMatches(matchingRequest);
```

Future API here could be:
```suggestion
        const hashResponse = await hashContent(file);
        // Result: { "pdq": "...", "md5": "..." }
        
        if (!hashResponse.ok()) {
          return; // error case
        }
        
        // Step 3: Use in matching API
        const matches = await findMatches(hashResponse);
```